### PR TITLE
[new-lair-5] lair api serialization types

### PR DESCRIPTION
**Pull request by neonphog**
DEPENDS ON https://github.com/holochain/lair/pull/63

[Video Walkthrough (1:34)](https://www.youtube.com/watch?v=IhXleu8RmLM)

Start with the [LairApiEnum](https://github.com/holochain/lair/pull/64/files#diff-501c62882899456a70a744e94f56daaa560643f1428f8851276b1bc540d64a58R86) for the list of requests/responses available in the lair api, then move onto the individual struct modules for details about the parameters,

Each api struct can be marked with the marker traits `AsLairRequest` or `AsLairResponse` to facilitate the type system knowing which type will be returned from a specific request.

### CODE REVIEW

File: crates/lair_keystore_api/src/lair_api/hello.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+}
+
+/// The hello response from the target lair instance.
+/// This data allows us to verify we are speaking to our expected target.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct LairApiResHello {
+    /// msg id to relate request / response.
+    pub msg_id: Arc<str>,
```
Comment (freesig) [inline crates/lair_keystore_api/src/lair_api/hello.rs:48]: Any reason these need to be an Arc?

**Review by freesig — APPROVED** [review]

File: crates/lair_keystore_api/src/lair_api/hello.rs (hunk 1 above)
Comment (neonphog) [reply crates/lair_keystore_api/src/lair_api/hello.rs:48]: They are cloned / used as hash map key, etc.

Comment (maackle) [reply crates/lair_keystore_api/src/lair_api/hello.rs:48]: Seems like these 3 fields together might be under 100 bytes. Wonder if cloning 100 bytes is significantly more overhead than 3 Arcs.

Comment (neonphog) [reply crates/lair_keystore_api/src/lair_api/hello.rs:48]: arcs only add 2 usizes of overhead, a single clone of an arc almost always validates its usage

**Review by maackle — APPROVED** [review]
Looks good. It's actually kind of nice that you didn't use a macro to generate all this boilerplate, even though this seems like it would have been an ideal place to do so. This stuff shouldn't need to change much.

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**